maldoca/ole/ppt.cc

```diff
@@ -185,7 +185,7 @@ StatusOr<std::string> VBAProjectStorage::Decompress(
   // Decompress zlib chunks
   ZLib z;
   int result = z.UncompressChunk(
-      reinterpret_cast<Bytef *>(&(storage[0])), &decompressed_size,
+      reinterpret_cast<Bytef *>(&(storage[0])), reinterpret_cast<uLongf *>(&decompressed_size),
```

Comment:
@david: any better/safer way to cast?

Reply:
not sure why you need this but you can always use a local variable of type uLongf to copy the decompressed_size into ( or you can just rename the parameter to something else and make a ULongf decompressed_size = ... ;)  Then at most you get a truncation error but not likely a problem ever.

Reply:
otherwise it results in "error: cannot initialize a parameter of type 'uLongf *' (aka 'unsigned long *') with an rvalue of type 'size_t *' (aka 'unsigned long long *')
      reinterpret_cast<Bytef *>(&(storage[0])), &decompressed_size,"

is there any relevant diff between the reinterpret_cast and the temp var solution?

Reply:
the problem is long != long long always. On 64 bit platform they ought to be but on 32 bit platform long is 32 bits. Then if you cast a 64 bit pointer to 32 bit pointer, it will work with little endian but not big endian.  So instead of worrying about that, let the system implicitly convert the number if necessary and in the worst case you get a truncation problem but you will not get a seg fault. In practice, it's unlikely to be a problem but might as well do it the right way.

Reply:
not sure if it's only related to 32 vs 64bit. it works on linux x64 but does not on winx64. hence, i assume there is also an os-dependent component.

anyways, moved the non path separator related code into a separate cl: https://github.com/google/maldoca/pull/51 as i am still investigating the path related issues.